test/unit/transports/aws_test.rb

```diff
@@ -90,5 +90,12 @@ def initialize(hash)
       creds[:credentials].session_token.must_equal 'test_session_token'
       creds[:region].must_equal 'test_region'
     end
+
+    it 'validate aws connection fallback' do
+      options[:profile] = nil
+      options[:access_key_id] = nil
+
+      Aws.config.expects(:update).never
```

Comment:
I don't think this test is doing anything. Don't you want to call `connection.connect`? And doesn't `Aws.config` always receive `:update` but in this case, it will be with options that don't include a `creds` key?